### PR TITLE
Volunteer Credits Table UI

### PR DESCRIPTION
**Pull request by mendelB**
### What's this PR do?

This pull request adds the UI for the new Volunteer Credits Table in the Profile Credits tab.

### How should this be reviewed?
Apologies! This is a bit of an involved UI and thus introduces a fair amount to review. I couldn't seem to finagle a feasible way to split this up.

Do the components and concerns seem contained logically? Does everything make sense? There's stubbed data for how we expect this to look, but this should give you a good idea of the two interfaces - mobile & tablet/desktop.

### Any background context you want to provide?
There are essentially two UI concerns here
- the table largely, for mobile/desktop
- the 'Campaign preview' node specifically for mobile/desktop.

These should both hopefully be smoothly responsive, while relatively nicely logically embedded together.

I'm not quite sure what to do about the overflow on certain viewports, specifically at medium but smallish size (see screenshots). It's set to `scroll`, but I imagine that's not optimal.

### Relevant tickets
[Abstract mocks](https://app.abstract.com/projects/fadf0790-f1f9-11e6-9a30-b54ba7e8e705/branches/0418693f-2501-46f7-8b32-7187f99b8b44/collections/76150b3e-29df-4d71-aa1a-b789e8ff539d)
References [Pivotal #171908706](https://www.pivotaltracker.com/story/show/171908706).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [x] Added screenshots of front-end changes on small, medium, and large screens.
- [ ] Added appropriate feature/unit tests.

Kapture 2020-03-24 at 15 57 13

- [Small](https://user-images.githubusercontent.com/12417657/77470916-d5333f80-6de7-11ea-9300-0b78ba763de1.png)
- [Medium (notice the overflow)](https://user-images.githubusercontent.com/12417657/77470931-d7959980-6de7-11ea-90ef-6602268449a8.png)
- [Large](https://user-images.githubusercontent.com/12417657/77470938-dbc1b700-6de7-11ea-9aba-0eac4c1f744d.png)

### CODE REVIEW

**Comment by weerd** [comment]
image

I wonder if when it gets to around this size, we could stack the image and corresponding campaign description. OR if not, maybe hide the image 🤷‍♂ since not as important.

**Comment by mendelB** [comment]
> I wonder if when it gets to around this size, we could stack the image and corresponding campaign description. OR if not, maybe hide the image 🤷‍♂ since not as important.

Ooh that's a good call! I can try and give that a whirl!

**Comment by mendelB** [comment]
Kapture 2020-03-25 at 15 15 42

@weerd employed your idea in https://github.com/DoSomething/phoenix-next/pull/1992/commits/a76b7f873f4fdd229e3a1e4cd1c76e38e7f0f607 and removed the `max-width` from this preview node, and it's not overflowing anymore! I think the issue now _might_ be that the rows get a bit too high when they're stacked at certain viewports, but I guess we can just remove the images entirely like your alternative idea! (CC @lkpttn any thoughts?)

**Comment by mendelB** [comment]
Hidden image:

Kapture 2020-03-25 at 15 19 18

**Comment by mendelB** [comment]
Enabled auth and this feature in the preview app!
https://dosomething-phoenix-de-pr-1992.herokuapp.com/us/account/profile/credits

**Comment by lkpttn** [comment]
I think the hidden image version is a little cleaner, I wish I knew a little bit more about how common some of those in-between resolutions were.

**Comment by weerd** [comment]
image

I'm seeing some auto scroll bar areas show up on the review app 🤔

**Comment by mendelB** [comment]
> image
> 
> I'm seeing some auto scroll bar areas show up on the review app 🤔

Oh wait for reals! huh we're not setting any hardcoded widths anymore, so I wonder why it'd do that 😭

**Comment by weerd** [comment]
Screen Shot 2020-03-26 at 11 07 25 AM

I have some visual recommendations; see the notes in screenshot. I also wonder if you should hold off on switch to the "table" view for a larger breakpoint so it doesn't need to be as awkward 🤔 

I've done this a bit with the new Homepage design where some transitions just were too compacted when switching to medium/large and so I held out longer for a larger breakpoint. Also been debating adding a couple, less common breakpoints at inbetween small --> medium & medium --> large for some of the edge cases...

**Comment by weerd** [comment]
> @weerd what viewport size are you seeing it in?

@mendelB it's weird... I see it on largest viewport but only AFTER I toggle off the responsive-mobile view in dev tools (still at largest viewport). But they do show in the default view.

**Comment by weerd** [comment]
image

It's this class/rule ^^

**Comment by DFurnes** [comment]
> @mendelB it's weird... I see it on largest viewport but only AFTER I toggle off the responsive-mobile view in dev tools (still at largest viewport). But they do show in the default view.

I think this might be macOS's disappearing scrollbars (which by default are hidden if using a touch device, but displayed if you're using a mouse). Regardless of your setting, they're hidden in "responsive design mode" to simulate a touch device.

**Comment by mendelB** [comment]
Thanks! I'm gonna take a look at implementing your visual notes @weerd and see if I can rid that scroll on larger viewports! (I realize the buttons _do_ have a fixed width so that might be it.) I had the overflow scroll rule set so that it'd be scroll-able when it overflowed, but of course, we'd like to prevent that in the first place 😬

**Comment by mendelB** [comment]
@weerd I finessed this a bit per your suggestions https://github.com/DoSomething/phoenix-next/pull/1992#issuecomment-604488786 in https://github.com/DoSomething/phoenix-next/pull/1992/commits/b9d2152d86647458c930d72fb695ec21fc8211c6, https://github.com/DoSomething/phoenix-next/pull/1992/commits/edc2703a25e58278cf804f0d6bce6273af08ab4c, and https://github.com/DoSomething/phoenix-next/pull/1992/commits/2d4566c6410717bfdc3d1dcb5d0787ef2ed2996c. I wasn't able to do too much about responsiveness on the smaller viewport to extend it past medium-size screens to avoid the table, since it doesn't stretch all that much, and my trials of moving things around didn't bear any reasonable fruit. But I _think_ it's at least looking way better now! ([Review app](https://dosomething-phoenix-de-pr-1992.herokuapp.com/us/account/profile/credits)).

File: resources/assets/components/pages/AccountPage/Credits/Credits.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      <SectionHeader underlined title="Volunteer Credits" />
+      <p>
+        Earn volunteer credits through volunteering. Your certificates will
+        appear here within 4 days of you completing a campaign, after our staff
+        is able to verify your work! Learn more about our volunteer credit
+        program.
+      </p>
+    </div>
+
+    <div className="grid-wide my-8 overflow-scroll">
```
Comment (weerd) [inline resources/assets/components/pages/AccountPage/Credits/Credits.js]: So this `overflow-scroll` class is still forcing the scroll bars that do nothing. Is there a reason you have to leave it in? I tested it with it toggled off and seems fine... 🤔

Comment (mendelB) [reply resources/assets/components/pages/AccountPage/Credits/Credits.js]: Gah, sorry I thought I took that out! _actually_ removed in https://github.com/DoSomething/phoenix-next/pull/1992/commits/709f6929d4a5ff798f679779f5dc89916f5b6359

**Comment by weerd** [comment]
Screen Shot 2020-03-27 at 1 00 24 PM

@mendelB if you disable the `max-w-md` class until the actual breakpoint it's needed, along with the `h-16` class on the image, you can get everything to stretch nicely across instead of being weirdly cut off to one side at around this size in the viewport 😄

**Comment by mendelB** [comment]
> Screen Shot 2020-03-27 at 1 00 24 PM
> 
> @mendelB if you disable the `max-w-md` class until the actual breakpoint it's needed, along with the `h-16` class on the image, you can get everything to stretch nicely across instead of being weirdly cut off to one side at around this size in the viewport 😄

Noice! Ok, I messed around a bit with adjusting the widths (on smaller & larger viewports we need that contained height on the image so it doesn't awkwardly stretch). Overall I think it's looking way sharper now! 
https://github.com/DoSomething/phoenix-next/pull/1992/commits/d26003fbaafdbb59f5ab468d923ab6f4eb399147, https://github.com/DoSomething/phoenix-next/pull/1992/commits/3950058a3e51652b97b74a8ba8ca5a11096718f1

**Comment by weerd** [comment]
> Overall I think it's looking way sharper now!

Ya! Looking really good 💃

File: resources/assets/components/pages/AccountPage/Credits/VolunteerCreditsTable.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    },
+  },
+  {
+    showcaseTitle: 'Are you ready to vote?',
+    showcaseDescription: "Take a quiz to see if you're ready to vote.",
+    showcaseImage: {
+      url: IMAGE_URLS[2],
+    },
+  },
+];
```
Comment (weerd) [inline resources/assets/components/pages/AccountPage/Credits/VolunteerCreditsTable.js]: Maybe add a comment just to help indicate this data is just for mocking stuff for now.

File: resources/assets/components/pages/AccountPage/Credits/VolunteerCreditsTableRow.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+};
+
+// The certificate PDF Download button with pending and ready states.
+const DownloadButton = ({ pending }) => (
+  <button
+    type="button"
+    css={css`
+      height: 65px;
+    `}
+    className={classNames('button w-full', { 'is-disabled': pending })}
```
Comment (weerd) [inline resources/assets/components/pages/AccountPage/Credits/VolunteerCreditsTableRow.js]: Not needed for merge, but did you try the new `btn` component class? Trying to move away from the Forge based classes.

File: resources/assets/components/pages/AccountPage/Credits/VolunteerCreditsTableRow.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+          <CampaignPreview campaignWebsite={campaignWebsite} />
+
+          <ul className="py-5">
+            <PostDetail detail="Action Type" value={actionType} />
+            <PostDetail detail="Volunteer Hours" value={volunteerHours} />
+            <PostDetail detail="Date Completed" value={dateCompleted} />
+          </ul>
+
+          <DownloadButton
+            pending={Boolean(Math.floor(Math.random() * Math.floor(2)))}
```
Comment (weerd) [inline resources/assets/components/pages/AccountPage/Credits/VolunteerCreditsTableRow.js]: 😏

File: resources/assets/components/pages/AccountPage/Credits/CampaignPreview.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  getFormattedScreenSize,
+} from '../../../../helpers';
+
+const CampaignPreview = ({ campaignWebsite }) => {
+  const { showcaseTitle, showcaseDescription, showcaseImage } = campaignWebsite;
+
+  const showcaseImageUrl =
+    getFormattedScreenSize() === 'small'
+      ? contentfulImageUrl(showcaseImage.url, 200, 100, 'fill')
+      : contentfulImageUrl(showcaseImage.url, 640, 360, 'fill');
```
Comment (weerd) [inline resources/assets/components/pages/AccountPage/Credits/CampaignPreview.js]: Not required for merge but maybe a later upgrade to let `srcset` handle this?

**Comment by mendelB** [comment]
> 👍 Looks good! I had a couple additional comments but nothing that blocks merging 🥇

🎊I left a few TODO comments for follow-up. Thanks, @weerd for your incredibly helpful feedback and guidance on this!!

**Pull request merged**